### PR TITLE
Code Clean Up + Create Room

### PR DESCRIPTION
**Pull request by JackyChiu**
## What

Cleaned up code into their own modules. Client now enters the chat, gets shown what rooms are available. Then joins `Default Room` if no other rooms are created. 
## TODO's
- Enter username and use that as your username
- Option to join/create a room

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,81 +1,48 @@
...
+        console.log(chatRooms)
+        if (chatRooms.length === 1) {
+            console.log(`Joined ${chatRooms[0]}`)
+            socket.emit('create', chatRooms[0])
+        }
+    // TODO: Select room or create room
+    Input.sendMessage(socket)
     }).catch((err) => {
-        console.log(err);
+        console.error('Error getting rooms', err)
```
Comment (BenEmdon) [inline app.js]: what's better about `console.error`?

Comment (JackyChiu) [reply app.js]: Opps it was a mistake but the benefit is that it would print to `stderr` instead of `stdout` but we don't use `stderr` so I'll remove it

**Comment by vikrambombhi** [comment]
everything looks good but idk how to use promises so someone else has to check that

**Comment by JackyChiu** [comment]
@VikramBombhi Dw my promises are always on point 😉

**Pull request merged**